### ISSUE TITLE
Named imports problem

### ISSUE THREAD

**Issue by Ky6uk** [Igorbek/typescript-plugin-styled-components#10]
Those two snippets generate different classnames. That behaviour breaks style extending feature.

```ts
// Base.style.tsx
import styled from 'styled-components';

export const Header = styled.div`
  color: green;
  font-weight: 400;
  font-size: 15px;
`;
```

---
* This is okay. `Header` and `H1` are different classnames. `H1` is an extended version of `Header`.

```ts
import styled from 'styled-components';
import { Header } from './Base.style';

export const H1 = styled(Header)`
  font-size: 20px;
`;
```
---
* This is **not** okay. Exported version of `Header` the same as imported (the same classname).

```ts
import styled from 'styled-components';
import { Header as BaseHeader } from './Base.style';

export const Header = styled(BaseHeader)`
  font-size: 20px;
`;
```

**Comment by Igorbek** [comment]
In general, you should not rely on the class names as they are generated by `styled-components` and encapsulated. This plugin only adds human-readable name to that names for simplifying debugging.

Can you please elaborate:
- what are the class names being generated in your case?
- what are expected class names?
- how does it break style extending feature?
- how are you using these names for style extending?

**Comment by Ky6uk** [comment]
There is the elementary example above which works fine without this plugin or without named imports. Just try to compile it **with** and **without** `typescript-plugin-styled-components` and you will see the output completely different because the plugin breaks SC-generated classnames.

* ✅ `SC` + named imports **without** `tpsc`.
* ✅ `SC` + `tpsc` **without** named imports.
* ❌ `SC` + `tpsc` **with** named imports.

There is the `same` code. I just toggle this plugin on and off.

#### Without plugin
![screen shot 2018-04-23 at 22 37 03](https://user-images.githubusercontent.com/46930/39149144-e27a3dc0-4746-11e8-8bc6-4748770b60dd.png)

As you can see `TEST` is the extended version of `TEST2`. Because a style cascading working fine.

> `sc-ifAKCX` is extending `sc-bxivhb`.
> `cnQDwC` is extending `dYnMnB`.
>
>  ![screen shot 2018-04-23 at 22 42 24](https://user-images.githubusercontent.com/46930/39149410-a38e9844-4747-11e8-83c7-5f4dc031e9a7.png)
>  ![screen shot 2018-04-23 at 22 49 19](https://user-images.githubusercontent.com/46930/39149760-e0d6bf8c-4748-11e8-87bc-c836ea2d9c38.png)



#### WIth plugin
![screen shot 2018-04-23 at 22 33 16](https://user-images.githubusercontent.com/46930/39148975-5cc114d8-4746-11e8-9755-49657e932d5f.png)

Transformed code just breaks the style cascading for no reasons.
 
> ![screen shot 2018-04-23 at 22 45 53](https://user-images.githubusercontent.com/46930/39149616-65c3013e-4748-11e8-8a99-85a75e44ecf7.png)

**Comment by Igorbek** [comment]
Filed an issue https://github.com/styled-components/styled-components/issues/1715

**Comment by Ky6uk** [comment]
It looks fixed in SC v3.3.3.
